faults.md

```diff
@@ -14,39 +14,81 @@ A fault is what happens when partcipants in the protocol are behaving incorrectl
-  - **Check:** The chain checks that the proof verifies with the missing sectors.
-  - **Penalization:** The miner is penalized for collateral and power proportional to the number of missing sectors. The sectors are also removed from the miners proving set.
-    - TODO: should the collateral lost here be proportional to the remaining time?
-    - TODO(nicola): check if the time between posting two proofs allows for a generation attack if it does not then we might reconsider the sector not being lost
-  - Note: if a sector is missed here, and they are recovered after the fact, the miner could simple 're-commit' the sector. They still have to pay the collateral, but the data can be quickly re-introduced into the system to avoid clients calling them out for breach of contract (this would only work because the sector commD/commR is the same)
-  - Note: In the case where a miner is temporarily unable to prove some of their data, they can simply wait for the temporary unavailability to recover, and then continue proving, submitting the proofs a bit late if necessary (paying appropriate fees, as described above).
-- **Breach of contract dispute:**
-  - **Condition:** A client who has stored data with a miner, and the miner removes the sector containing that data before the end of the agreed upon time period.
-  - **Reporting:** The client invokes `ArbitrateDeal` on the offending miner actor with a signed deal from that miner for the storage in question. Note: the reporting must happen within one proving period of the miner removing the storage erroneously.
-  - **Check:** The chain checks that the deal was correctly signed by the miner in question, that the deal has not yet expired, and that the sector referenced by the deal is no longer in the miners proving set.
-  - **Penalization:** The miner is penalized an amount proportional to the incorrectly removed sector. This penalty is taken from their pledged collateral .
-  - Note: This implies that miners cannot re-seal data into different sectors. We could come up with a protocol where the client gives the miner explicit consent to re-seal, but that is more complicated and can be done later.
+
+**Definitions**
+
+- `PoStProvingPeriod: UInt = 24 * 60 * 60 / 30 = 2880` (1 day, given `30s` block time)
+- `PoStTimeout: UInt` (less than 1 ProvingPeriod)
+- `SectorFailureTimout: UInt` (more than 1 Proving Period)
+
+TODO: define the above constants
+
+#### PoSt Late Submission
+
+- **After `ProvingPeriodEnd`:**
+  - **Condition**: If the miner posts their PoSt after the proving period ends, but before `PoStTimeout`
+  - **Reporting:** The miner submits their PoSt as usual, but includes the `LateSubmissionFee`.
+  - **Penalization:**
+    - *Economic penalization*: To determine the penalty amount, `ComputeLateSubmissionFee(minerPower)` is called.
+    - *Power penalization*: The miners' power is reduced to `0`.
```

Comment:
> This is a big change. Previously, a miner maintained power when late, until the generation attack threshold was exceeded (which made sense to me)

I don't think that was the case, but rather it was not well specified.

Reply:
> But I'm also confused about when it should take effect.

See https://github.com/filecoin-project/specs/pull/384 for some details on how this will be enforced in practice.

Reply:
> This is an additional complexity, since the Miner.Power state variables will no longer define the power table; it will be a function of more complex state 

This is already the case with the previous definition, as slashed miners would have had their power removed, even without anyone calling `SlashStorageMiner`

Reply:
@dignifiedquire 

> See #384 for some details on how this will be enforced in practice.

I read #384 and it's still not clear to me when the miner's power is reduced to 0. Should `GetMinersPowerAt` return 0 when the head of the chain reaches the height which marks the end of the miner's proving period [and they have yet to submit a PoSt for that proving period]?

Reply:
I can now happily answer that yes it should. That is, if the miner's PoSt has not been included by the block at height ProvingPeriodEnd, the miner's power will read 0 (they will be unable to participate in leader election) until they can prove they are storing the sectors to which they committed.